src/JustEat.StatsD/StatsDMessageFormatter.cs

```diff
@@ -91,7 +92,7 @@ public string Increment(long magnitude, double sampleRate, string statBucket)
                 return _prefix + statBucket + ":1|c";
             }
 
-            var stat = string.Format(InvariantCulture, "{0}{1}:{2}|c", _prefix, statBucket, magnitude);
+            var stat = string.Concat(_prefix, statBucket, ":", magnitude.ToString(InvariantCulture), "|c");
```

Comment:
Could the reason for the allocation be that this overload is a `params string[]`, so there's an array being allocated? Looking at the overloads on MSDN, there's only overloads up to 4 objects before it becomes `params object[]`.

Reply:
Could be. But using multiple `string.Concat` doesn't make it better (184B allocated), so this might be as good as it can be.

Reply:
What's actually in the compiled binary for this code if you look at it with ILSpy/Reflector/etc?

Reply:
yes it generates IL like
````
 newarr       [mscorlib]System.String
...
 string [mscorlib]System.String::Concat(string[])
````

Using 2 `string.Concat` calls has worse perf and worse allocation, so this might be as good as it can be.

Reply:
Well that's at least the allocation mystery solved then.